### ISSUE TITLE
Array.from doesn't work on Sets in chrome.

### ISSUE THREAD

**Issue by AdaRoseCannon** [polyfillpolyfill/polyfill-service#420]
```
Array.from(new Set([1,2]));
[]
```

**Comment by triblondon** [comment]
The MDN polyfill also fails on this case, but the native implementation in Firefox works.  It's tough to make this work without other parts of ES6.  Eg you could do:

```
var array = [...set];
```

**Comment by triblondon** [comment]
@JakeChampion I was thinking about this one this morning.  Our Array.from implementation assumes array-like things are ES5 constructs, and normalises them to an object with a length and elements as properties.  Whereas the [core-js version](https://github.com/zloirock/core-js/blob/master/modules/es6.array.from.js) also checks whether the object is iterable.  Can we do that or would we need to polyfill iterators to make that work?  What do you think?

**Comment by JakeChampion** [comment]
Maybe this?

``` js
if (typeof Set === "function" && source instanceof Set) {
    var a = source.values();
    var i;
    var array = [];
    var length;
    for (i = 0; ; ++i) {
        var next = a.next();
        if (next.done) {
            break;
        }
        var value = next.value;
        array[i] = value;
        length = i;
    }
}
```

**Comment by JakeChampion** [comment]
We can avoid the dependency of an iterator polyfill if we only care about the standard iterable objects. If we want `Array.from` to work with custom iterables then we would need `Array.from` to be aware of iterable objects.

**Comment by zloirock** [comment]
@JakeChampion too many special cases. For `Set`, `Map`, generators - ok, but strings - should iterate over code points, generic iterables...

**Comment by JakeChampion** [comment]
Note - [es6-shim](https://github.com/paulmillr/es6-shim/blob/b81c0880dad3bcd7d313a40049fe3009babc3fb9/es6-shim.js#L702) supports iterables.

**Comment by JakeChampion** [comment]
@zloirock You raise a good point about iterating over strings. Normal strings would be easy enough. Strings containing surrogate pairs would require some extra effort, [mathiasbynens/String.prototype.at](https://github.com/mathiasbynens/String.prototype.at) should resolve that issue.

Do you think catering to the standard iterable objects would require more code in the shim than making `Symbol` a dependency? _I'm thinking it would_.

**Comment by zloirock** [comment]
@JakeChampion I use [this common code](https://github.com/zloirock/core-js/blob/v2.0.3/modules/_string-at.js) for `String#at`, `String#codePointAt` and `String#@@iterator`.

In `core-js` I don't add `Symbol` polyfill as dependency to features working with iterables and use internal keys for well-known symbols.

**Comment by JakeChampion** [comment]
Work for this is being done in #665

**Comment by triblondon** [comment]
Fixed by #665
